src/ceph-disk

```diff
@@ -2183,15 +2211,16 @@ def main_activate(args):
             raise Error('%s is not a directory or block device' % args.path)
 
         if args.mark_init == 'none':
-            command_check_call(
-                [
-                    'ceph-osd',
-                    '--cluster={cluster}'.format(cluster=cluster),
-                    '--id={osd_id}'.format(osd_id=osd_id),
-                    '--osd-data={path}'.format(path=osd_data),
-                    '--osd-journal={path}/journal'.format(path=osd_data),
-                ],
-            )
+            cmd = [
+                'ceph-osd',
+                '--cluster={cluster}'.format(cluster=cluster),
+                '--id={osd_id}'.format(osd_id=osd_id),
+                '--osd-data={path}'.format(path=osd_data)
+            ]
+            (allows_journal, wants_journal, needs_journal) = check_journal_reqs(args)
+            if wants_journal:
+                cmd.append('--osd-journal={path}/journal'.format(path=osd_data))
```

Comment:
these aren't really needed here since the default config values should work.  they are needed for --mkfs because we're in a temp directory (not /var/lib/ceph/osd/$cluster-$id) tho.  but this a bit safer as is i guess!